### ISSUE TITLE
Reaches do not connect to wastewater_nodes

### ISSUE THREAD

**Issue by urskaufmann** [QGEP/QGEP#443]
QGEP Plugin 1.1.1
If I create a new reach, there is no automatic connection to the wastewater_node.
There is also an error if I want connect with the connect wastewater networkelements:
![error connecting](https://user-images.githubusercontent.com/1257403/44383736-ffd95a00-a519-11e8-97ea-11ded95e65d1.PNG)

**Comment by urskaufmann** [comment]
I tried to connect with the default value as recommended in #361 , but was not successful. I got the obj_id of the wastewater_structure in the rp_to or from_fk_wasterwater_networkelement. And therefore an error when saving. I replaced obj_id with wn_obj_id, but did not help.
Is this solution able to handle the connection to an other reach (only for rp_to!), if there is no ww-node?
And next point: because I want to work with the option "reuse last attribut values": if I digitize a reach without snapping, there was the last rp_*_fk_wastewater_networkelement -value in the field and in this case it should be NULL.

**Comment by urskaufmann** [comment]
the above expression works when snapping to reaches or vw_wastewater_node, if you change wn_obj_id to obj_id...
That works if I use the snapping options.
![unbenannt](https://user-images.githubusercontent.com/1257403/45253897-9e651800-b36e-11e8-8121-e12d2d336ad9.PNG)
 But the reach-layer is stronger than the wastewater_node-layer. Can I change this? Seems that reach-layer is stronger, because it's the layer that I edit.

**Comment by urskaufmann** [comment]
Better version of the expression:
add the following expression to the default value for rp_from_fk_wastewater_networkelement:

```
with_variable(
   'snapped_result',
   array_first(@snapping_results), if(map_get(@snapped_result,'valid')=true, 
   attribute(get_feature_by_id(map_get(@snapped_result, 'layer'), map_get(@snapped_result, 'feature_id')), 'obj_id'),'NULL'))
```
   
Change array_first to array_last for rp_to_fk_wastewater_networkelement

With this expression, there is no fk, if there is no snapping (also when the option reuse attribute values is on).
Requirement: Snapping options are set as shown in the post above.
If you digitize careful, you can set the fk automatically to the node you want (also if there are two nodes in the wastewater_structure) or to the reach.

Improvement proposal: 
- snapping to node should always be preferred if there is a node. Or the described snapping options should be set automatically, if I digitize new reaches.
- there should by a fk to a reach only for a reachpoint_to, but not for a reachpoint_from (Erfassungsgrundsätze VSA).

**Comment by urskaufmann** [comment]
@sjib @m-kuhn Priority 2019 Bugfix
Is the wizard necessary or not?

**Comment by m-kuhn** [comment]
# Situation

It's currently almost impossible to link a reach to a wastewater node to which another reach is already connected. Often (always?) the snapping will happen to the reach.

It would be nice to be able to completely do this with native QGIS tools in the long run, but there are a lot of edge cases before it's possible to get rid of the wizard.

# Approach

As a short term solution fix the wizard

**Comment by ponceta** [comment]
According to 03.06.2019 discussion

Connect tool works (fixed with plugin 1.1.10)

Digitize does not (should be fixed)

**Comment by urskaufmann** [comment]
Is fixed with Plugin 1.1.11
